lib/cisco_node_utils/bgp_neighbor_af.rb

```diff
@@ -579,6 +618,47 @@ def send_community=(val)
       config_set('bgp_neighbor_af', 'send_community', @set_args)
     end
 
+    def send_comm_ios_xr_set(val)
+      if val[/none/]
+        state = 'no'
+        val = 'both'
+      end
+      if val[/extended|standard/]
+        case send_community
+        when /both/ # legacy command on Nexus
+          state = 'no'
+          # Unset the opposite property
+          val = val[/extended/] ? 'standard' : 'extended'
+        when /extended|standard/
+          # This is an additive property therefore remove the entire command
+          # when switching from: ext <--> std
+          if val == 'standard'
+            set_args_keys(state: 'no', attr: 'send-community-ebgp')
+          else # val == 'extended'
+            set_args_keys(state: 'no', attr: 'send-extended-community-ebgp')
+          end
+          config_set('bgp_neighbor_af', 'send_community', @set_args)
+          state = ''
+        end
+      end
+      if val == 'both'
```

Comment:
I find the flow of this function quite confusing due to the intermingling of if, else, when. Would the following work instead?

```ruby
def send_comm_ios_xr_set(val)
  case val
  when 'none'
    set_args_keys(state: 'no', attr: 'send-community-ebgp')
    config_set('bgp_neighbor_af', 'send_community', @set_args')
    set_args_keys(state: 'no', attr: 'send-extended-community-ebgp')
    config_set('bgp_neighbor_af', 'send_community', @set_args')
  when 'standard'
    set_args_keys(state: '', attr: 'send-community-ebgp')
    config_set('bgp_neighbor_af', 'send_community', @set_args')
    set_args_keys(state: 'no', attr: 'send-extended-community-ebgp')
    config_set('bgp_neighbor_af', 'send_community', @set_args')
  when 'extended'
    set_args_keys(state: 'no', attr: 'send-community-ebgp')
    config_set('bgp_neighbor_af', 'send_community', @set_args')
    set_args_keys(state: '', attr: 'send-extended-community-ebgp')
    config_set('bgp_neighbor_af', 'send_community', @set_args')
  when 'both'
    set_args_keys(state: '', attr: 'send-community-ebgp')
    config_set('bgp_neighbor_af', 'send_community', @set_args')
    set_args_keys(state: '', attr: 'send-extended-community-ebgp')
    config_set('bgp_neighbor_af', 'send_community', @set_args')
  end
end
```